src/PhotosphereEditor.tsx

```diff
@@ -118,6 +124,69 @@ function PhotosphereEditor({
       );
     return null;
   }
+
+  // Function to handle name change
+  function handleNameChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setNewName(event.target.value);
+  }
+
+  // Function to handle background change
+  function handleBackgroundChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const file = event.target.files?.[0];
+    if (file) {
+      setNewBackground(URL.createObjectURL(file));
+    }
+  }
+
+  // Function to handle submit button click for name change
+  function handleSubmitName() {
+    if (newName.trim() !== "") {
+      const currentPhotosphere = vfe.photospheres[currentPS];
+      //making updated photosphere list minus the currentPS
+      const updatedPhotospheres: Record<string, Photosphere> =
+        Object.fromEntries(
+          Object.entries(vfe.photospheres).filter(([key]) => key !== currentPS),
+        );
+      //making currentPS entry with newName
+      updatedPhotospheres[newName] = { ...currentPhotosphere, id: newName };
```

Comment:
This function also needs to update any `PhotosphereLink` hotspots because other photospheres may be linking to the current one.